pkg/api/validate/openshiftcluster_validatedynamic_test.go

```diff
@@ -148,3 +153,117 @@ func TestValidateProviders(t *testing.T) {
+
+	for _, tt := range []struct {
+		name    string
+		mocks   func(*mocknetwork.MockVirtualNetworksClient)
+		oc      *api.OpenShiftCluster
+		wantErr string
+	}{
+		{
+			name: "fail: custom dns set",
+			mocks: func(networksClient *mocknetwork.MockVirtualNetworksClient) {
+				networksClient.EXPECT().
+					Get(gomock.Any(), resourceGroup, vnetName, "").
+					Return(mgmtnetwork.VirtualNetwork{
+						VirtualNetworkPropertiesFormat: &mgmtnetwork.VirtualNetworkPropertiesFormat{
+							DhcpOptions: &mgmtnetwork.DhcpOptions{
+								DNSServers: &[]string{
+									"172.16.1.1",
+								},
+							},
+						},
+					}, nil)
+			},
+			wantErr: "custom DNS servers are not supported.",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					MasterProfile: api.MasterProfile{
+						SubnetID: validSubnet,
+					},
+				},
+			},
+		},
+		{
+			name: "pass: default settings",
+			mocks: func(networksClient *mocknetwork.MockVirtualNetworksClient) {
+				networksClient.EXPECT().
+					Get(gomock.Any(), resourceGroup, vnetName, "").
+					Return(mgmtnetwork.VirtualNetwork{
+						VirtualNetworkPropertiesFormat: &mgmtnetwork.VirtualNetworkPropertiesFormat{
+							DhcpOptions: &mgmtnetwork.DhcpOptions{
+								DNSServers: &[]string{},
+							},
+						},
+					}, nil)
+			},
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					MasterProfile: api.MasterProfile{
+						SubnetID: validSubnet,
+					},
+				},
+			},
+		},
+		{
+			name: "fail: vnet not found",
+			mocks: func(networksClient *mocknetwork.MockVirtualNetworksClient) {
+				networksClient.EXPECT().
+					Get(gomock.Any(), resourceGroup, vnetName, "").
+					Return(mgmtnetwork.VirtualNetwork{}, fmt.Errorf("The Resource 'Microsoft.Network/virtualNetworks/testVnet' under resource group 'testGroup' was not found."))
+			},
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					MasterProfile: api.MasterProfile{
+						SubnetID: validSubnet,
+					},
+				},
+			},
+			wantErr: "The Resource 'Microsoft.Network/virtualNetworks/testVnet' under resource group 'testGroup' was not found.",
+		},
+		{
+			name: "fail: invalid subnet",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					MasterProfile: api.MasterProfile{
+						SubnetID: "/invalid/subnet",
+					},
+				},
+			},
+			wantErr: "subnet ID \"/invalid/subnet\" has incorrect length",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			networkClient := mocknetwork.NewMockVirtualNetworksClient(controller)
+
+			if tt.mocks != nil {
+				tt.mocks(networkClient)
+			}
+
+			err := dv.validateVnet(ctx, networkClient, tt.oc)
+			if err != nil && err.Error() != tt.wantErr {
+				switch e := err.(type) {
+				case *api.CloudError:
+					if !strings.HasSuffix(e.CloudErrorBody.Message, tt.wantErr) {
+						t.Error(err.Error())
+					}
+				default:
+					t.Errorf(err.Error())
+				}
+			}
```

Comment:
```suggestion
			if err != nil && err.Error() != tt.wantErr ||
				err == nil && tt.wantErr != "" {
				t.Error(err)
			}
```

use this standard pattern ^ (you'll need to update some error strings above as a consequence)

Reply:
The problem I encountered when using the standard pattern as referenced was that I was unable to get the exact `tt.wantErr` string to match the actual error string.  I believe this is due to  the error string looking like this:
```
/home/kwoodson/go/src/github.com/Azure/ARO-RP/pkg/api/validate/openshiftcluster_validatedynamic_test.go:261: 400: InvalidLinkedVNet: : The provided vnet '/subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/testGroup/providers/Microsoft.Network/virtualNetworks/testVnet' is invalid: custom DNS servers are not supported.
```

When I copied and pasted this error string into the `tt.wantErr` it would not match with the standard pattern.  Is that due to the stack being printed?  Is that due to the error being wrapped into a `api.CloudError` which also prints the statuscode `400` as well as the error string `InvalidLinkedVNet`?  I attempted multiple iterations and could not satisfy the pattern. 
 For this reason I modified the standard pattern to test for the CloudError.  Maybe there was something that I missed.  

I will attempt a couple more times.

Reply:
Looks like this worked as previous attempts did not.  Will update and resubmit.